bebraven/platform
# Update Peer Review assignment LTI launch URL on new course launch

### Description

**Task**: https://app.asana.com/0/1174274412967132/1199347615991597

### Motivation and Context

When we launch a new course, we copy over existing assignments in Canvas. Some of these assignments have LTI launch URLs that depend on the `Course` ID, which is different between the source and copied courses in our database. Specifically, assignments for projects, surveys, and peer evaluations have course-dependent routes at the moment. This PR handles updating the peer evaluation assignment's LTI launch URL when it's copied in `app/services/initialize_new_course.rb`. 

Adjusting the LTI launch URL goes away when we implement static endpoints for assignments:  https://app.asana.com/0/1174274412967132/1199352155608256

### Test

- spec, minimal based on stand-up discussion because this code is going away with static endpoints
- Manually create a course from a template and check that LTI launch URL
- Edit an existing Course and add a Peer Review assignment to it:
<img width="1792" alt="Screen Shot 2021-01-06 at 7 21 30 PM" src="https://user-images.githubusercontent.com/62911934/103847279-b808ee00-5054-11eb-9fc6-bc1f89df4226.png">

- Go the Course's `#show` page and click **Create Template**
<img width="1792" alt="Screen Shot 2021-01-06 at 7 21 37 PM" src="https://user-images.githubusercontent.com/62911934/103847314-d0790880-5054-11eb-8766-fb05e62d547d.png">

- Create the new Course, note the old course ID is 3:
<img width="1792" alt="Screen Shot 2021-01-06 at 7 21 44 PM" src="https://user-images.githubusercontent.com/62911934/103847362-ebe41380-5054-11eb-9064-abdc40f772e3.png">

- View new course on Canvas, view Assignments, click **Edit** for the Peer Evaluation assignment and click **More Options**:
<img width="1792" alt="Screen Shot 2021-01-06 at 7 22 21 PM" src="https://user-images.githubusercontent.com/62911934/103847447-19c95800-5055-11eb-8646-6c5bd3a4730d.png">

- Check the URL in the assignment, note the new course ID is 7 (and not 3) in the LTI launch URL:
<img width="1792" alt="Screen Shot 2021-01-06 at 7 22 27 PM" src="https://user-images.githubusercontent.com/62911934/103847487-349bcc80-5055-11eb-8072-ffc1066ebc6f.png">

### Types of changes




### Checklist:
